learners/WhenLearner.py

```diff
@@ -299,11 +358,25 @@ def predict(self, X):
 parameters_sgd = {'loss': 'perceptron'}
 
 
-def get_when_learner(name):
-    return WHEN_LEARNERS[name.lower().replace(' ', '').replace('_', '')]
+#######-------------------UTILITIES----------------#######
+
+
+def get_when_agent(name,**learner_kwargs):
```

Comment:
What is a when learner agent in comparison to a when learner? I'm a bit wary of the "agent" part of the name.

I was also thinking maybe it would make more sense for these "get_X_learner" functions to be renamed to like "import_X_learner" or something because they're mainly a way to grab a learner given just a string.

Reply:
Started an issue
https://github.com/cmaclell/apprentice_learner_api/issues/30